cognite/seismic/protos/v1/seismic_service_datatypes.proto

```diff
@@ -19,6 +19,37 @@ message LineSelect {
     }
 }
 
+// Correlated grid indices and coordinates
+message DoubleTraceCoordinates {
+    int32 iline = 1;
+    int32 xline = 2;
+    float x = 3;
+    float y = 4;
+}
+
+// Specify the grid by an origin point and the inline / crossline directions
+message OriginDirections {
+    // A point in the grid
+    DoubleTraceCoordinates origin = 1;
+    // The change in coordinates when increasing iline by 1
+    Coordinate iline_step = 2;
+    // The change in coordinates when increasing xline by 1
+    Coordinate xline_step = 3;
+}
+
+// Specify the grid by giving the coordinates of three or more corners
+message TraceCorners {
+    repeated DoubleTraceCoordinates corners = 1;
+}
+
+// Specify the affine transformation between line indices and coordinates
+message GridSpecifier {
+    oneof specifier {
```

Comment:
https://community.opengroup.org/osdu/data/data-definitions/-/blob/master/E-R/abstract/AbstractBinGrid.1.0.0.md - could we steal some naming, etc, from here?

Reply:
Hmm, it feels a bit of a clumsy way to specify...